server/portal/account.py

```diff
@@ -37,10 +56,10 @@ def login():
     if any(x is None for x in [email, password]):
         return jsonify({'error': 'Parameter is missing'}), 400
 
-    pw_hash = bcrypt.generate_password_hash(password)
-    account = PostAccount.query.filter(PostAccount.email == email and bcrypt.check_password_hash(pw_hash,
-                                                                                                 password)).first()
-    if account:
+    account = PostAccount.query.filter(PostAccount.email == email).first()
+    pw_hash = bcrypt.check_password_hash(account.encrypted_password, password)
```

Comment:
Check ```if account:```. We can't assume the user passes a valid email. Also, change ```pw_hash``` to a more descriptive variable name. Maybe ```is_correct_password```?

Reply:
Was already checking for valid email. Renamed password check.